packages/application-shell/src/test-utils/test-utils.js

```diff
@@ -280,41 +283,53 @@ const renderApp = (
   const mergedProject = mergeOptional(defaultProject, project);
   const mergedEnvironment = mergeOptional(defaultEnvironment, environment);
   const mergedGtmTracking = mergeOptional(defaultGtmTracking, gtmTracking);
-  return {
-    ...rtl.render(
-      <IntlProvider locale={locale}>
-        <ApolloProviderComponent mocks={mocks} addTypename={addTypename}>
-          <ConfigureFlopFlip
-            adapter={adpater}
-            defaultFlags={flags}
-            adapterArgs={defaultFlopflipAdapterArgs}
-          >
-            <ApplicationContextProvider
-              user={mergedUser}
-              project={
-                mergedProject && {
-                  ...mergedProject,
-                  allAppliedPermissions: denormalizePermissions(permissions),
-                  allAppliedActionRights: denormalizeActionRights(actionRights),
-                  allAppliedDataFences: denormalizeDataFences(dataFences),
-                }
+
+  const Wrapper = ({ children }) => (
```

Comment:
Nit: maybe call it `ApplicationProviders` otherwise this becomes fairly nifty in the bottom as it's "wrapper the wrapper if needed with the wrapper's wrapper" and you go 😑.